processing/src/main/java/io/druid/query/Queries.java

```diff
@@ -55,6 +61,18 @@ public static void verifyAggregations(
             "Missing fields [%s] for postAggregator [%s]", missing, postAgg.getName()
         );
         Preconditions.checkArgument(combinedAggNames.add(postAgg.getName()), "[%s] already defined", postAgg.getName());
+
+        if (postAgg instanceof HasDependentAggFactories) {
```

Comment:
@nishantmonu51 It makes sense to me, I will add `setDependentAggFactories` in `PostAggregator` and maybe I guess we would probably need another contextual information later and think about it then.